### PR TITLE
#162948921 Users should be able to tag their articles

### PR DESCRIPTION
**Pull request by boltonotieno**
#### What does this PR do?
Creates an option for users include tags in articles they author

#### Description of Task(s) to be completed?
* Create a tags model
* Link tags model with articles models

#### How should this be manually tested?
* Clone ah-alpha repo `git clone https://github.com/andela/ah-alpha.git`
* Checkout to the branch `ft-tag-articles-162948921`
* Install requirements `pip3 install -r requirements.txt`
* Run the server `python3 manage.py runserver`
* Launch postman and create a user by sending a POST request to `http://127.0.0.1:8000/api/v1/users/` with a body of:
```
{
   "user":{
        "username": "username",
        "email": "valid@email.com",
        "password": "Passw0rd"
    }
}
```
* Login by sending a POST request to `http://127.0.0.1:8000/api/v1/users/login/` with your registered email and password
```
{
   "user":{
        "email": "valid@email.com",
        "password": "Passw0rd"
    }
}
```
* Copy the returned JWT token and store it somewhere
* Use postman to create an article by sending a POST request to `http://127.0.0.1:8000/api/v1/articles`
* The body should have the format of 
```
{ 
  "article":{
       "image_path":"path",
            "title":"Title",
            "body":"Article body"
             "tags":"tag"
    }
}
```
* Select Bearer Token under Authorization and paste in the token gotten previously.
* The response should be similar to 
```
{
    "id": 14,
    "author": {
        "email": "ulumadan2@gmail.com",
        "username": "newuser"
    },
    "image_path": "dan",
    "slug": "rick-5",
    "title": "rick",
    "body": "morty",
    "tags": [
                "Tag" 
     ],
    "created_at": "2019-01-23T14:51:16.512539Z",
    "updated_at": "2019-01-23T14:51:16.512733Z",
    "favourites": false
}
```
* To Update the Tag, send a `PUT` request to `http://127.0.0.1:8000/api/v1/articles/<slug>/`
* The slug value is from the reponse you get when creating an article
* The body should have the format of 
```
{
	"tags": ["update"]
}
```
* The response should be similar to 
```
{
    "id": 14,
    "author": {
        "email": "ulumadan2@gmail.com",
        "username": "newuser"
    },
    "image_path": "dan",
    "slug": "rick-5",
    "title": "rick",
    "body": "morty",
    "tags": [
                "Update" 
     ],
    "created_at": "2019-01-23T14:51:16.512539Z",
    "updated_at": "2019-01-23T14:51:16.512733Z",
    "favourites": false
}
```
* Take get all tags, send a `GET` request to request to `http://127.0.0.1:8000/api/v1/tags/`
* The response should be similar to 

```
{
    "message": "Tags retrived Successfully",
    "tags": [
        "some tag",
        "another tag",
    ]
}
```
#### Any background context you want to provide?
* This project uses the latest version of python3. Please download it from [here](https://www.python.org/downloads/)
* You may as well use pip instead of pip3 if you have either only python3 installed on your computer or you have it set as the default version of python

#### What are the relevant pivotal tracker stories?
[#162948921](https://www.pivotaltracker.com/story/show/162948921)

### CODE REVIEW

File: authors/apps/articles/relations.py [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+from rest_framework.exceptions import ValidationError
+
+
+class TagsRelation(serializers.RelatedField):
+    """This class overwrites the serializer class for tags
+    to enable tags to be saved on a separate table when
+    creating an article
+    """
+
+    def get_queryset(self):
```
Comment (Ruiru11) [inline authors/apps/articles/relations.py:14]: for consistency include comments for this as well and the one below

Comment (boltonotieno) [reply authors/apps/articles/relations.py:14]: Okay, thanks. Let me work on that

Comment (boltonotieno) [reply authors/apps/articles/relations.py:14]: I have worked on it. You can kindly review and resolve the conversation

**Review by jamesbeamie — APPROVED** [review]

**Review by danuluma — APPROVED** [review]

**Review by danuluma — APPROVED** [review]
LGTM

**Review by frankip — APPROVED** [review]
LGTM

**Review by Ruiru11 — APPROVED** [review]

**Pull request merged**